wpengine/faustjs
# Replace `origin` with `protocol` + `host` in Faust sitemaps

## Description

This PR replaces the [`origin`](https://developer.mozilla.org/en-US/docs/Web/API/URL/origin) URL piece with [`protocol`](https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol) + [`host`](https://developer.mozilla.org/en-US/docs/Web/API/URL/host) as `origin` may resolve to a local container (such as `localhost:8080`) on hosting providers.



## Related Issue(s):

- #893

## Testing

1. From the monorepo root, checkout the PR, and run `npm run build`
2. Add the following to `examples/next/getting-started/src/pages/_middleware.ts`:
   ```ts
    import { handleSitemapRequests } from "@faustjs/next/middleware";
    import { NextRequest, NextResponse } from "next/server";
    
    export default async function _middleware(req: NextRequest) {
      const sitemapRequest = await handleSitemapRequests(req, {
        wpUrl: process.env.NEXT_PUBLIC_WORDPRESS_URL,
        sitemapIndexPath: `/wp-sitemap.xml`,
        pages: [
          {
            path: "/about",
          },
        ],
        async robotsTxt(sitemapUrl) {
          return `
            User-agent: *
            Allow: /
    
            Sitemap: ${sitemapUrl}
          `;
        },
      });
    
      if (sitemapRequest) {
        return sitemapRequest;
      }
    
      return NextResponse.next();
    }
   ```
3. Visit `http://localhost:3000/wp-sitemap.xml` and navigate to the index page, sitemap pages, robots.txt page, and faust pages sitemap to confirm the url replacement is happening for `http://localhost:3000`.
